pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -134,3 +140,78 @@ func (m DIDCommMsgMap) Decode(v interface{}) error {
 
 	return decoder.Decode(m)
 }
+
+func toMap(v interface{}) map[string]interface{} {
+	res := make(map[string]interface{})
+
+	rv := reflect.Indirect(reflect.ValueOf(v))
+	for rfv, field := range mapValueStructField(rv) {
+		// TODO: should be snake_case
+		name := strings.ToLower(field.Name)
+
+		tags := strings.Split(field.Tag.Get(`json`), ",")
+		if tags[0] != "" {
```

Comment:
does this happen when field.Tag.Get, itself returns an empty string?

Reply:
for now, the first one is enough, but I think we will handle `omitempty` as well

Reply:
I've just added omitempty as well